nuxeo-core/nuxeo-core-mongodb/src/test/java/org/nuxeo/ecm/core/mongodb/configuration/TestMongoDBBackEnd.java

```diff
@@ -0,0 +1,161 @@
+ * {@link CoreFeature} configure / load {@link org.nuxeo.ecm.core.test.StorageConfiguration} we should find a way to
+ * deploy others custom contribution {@link org.nuxeo.ecm.core.storage.mongodb.MongoDBRepositoryService} to easily
+ * enable / disable the contraints name checking in Tests .... perhaps we need to implement merge on the contribution
+ * 
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ MongoDBFeature.class, CoreFeature.class })
+@Deploy("org.nuxeo.ecm.core.mongodb.test:OSGI-INF/mongodb-disable-renaming-listener-test.xml")
+public class TestMongoDBBackEnd {
+
+    protected static final String DOCUMENT_NAME = "anyFile";
+
+    @Inject
+    protected CoreSession coreSession;
+
+    @Test
+    public void shouldCreateUniqueIndex() {
+        MongoDBConnectionService mongoService = Framework.getService(MongoDBConnectionService.class);
+        MongoDatabase database = mongoService.getDatabase(REPOSITORY_CONNECTION_PREFIX + "test");
+        MongoCollection<Document> collection = database.getCollection("test");
+        Optional<Document> index = StreamSupport.stream(collection.listIndexes().spliterator(), false)
+                                                .filter((d) -> d.get("key", Document.class).containsKey(KEY_PARENT_ID)
+                                                        && d.get("key", Document.class).containsKey(KEY_NAME))
+                                                .findAny();
+
+        assertTrue(index.isPresent());
+        Document document = index.get();
+        assertNotNull(document);
+        assertTrue((Boolean) document.get("unique"));
+    }
+
+    @Test
+    public void shouldFailsWhenCreateExistingChildNameDocument() {
+        DocumentModel domain = coreSession.createDocumentModel("/", "Folder", "Folder");
+        domain = coreSession.createDocument(domain);
+        addPermissions(domain, "james");
+        addPermissions(domain, "bob");
+        coreSession.save();
+
+        try (CloseableCoreSession bobSession = CoreInstance.openCoreSession(coreSession.getRepositoryName(), "bob")) {
+            createDocument(bobSession, domain, DOCUMENT_NAME);
+        }
+
+        try (CloseableCoreSession jamesSession = CoreInstance.openCoreSession(coreSession.getRepositoryName(),
+                "james")) {
+            createDocument(jamesSession, domain, DOCUMENT_NAME);
+            fail("should throw a ConcurrentUpdateException");
+        } catch (ConcurrentUpdateException cue) {
+            assertEquals(SC_CONFLICT, cue.getStatusCode());
+            assertTrue(
+                    cue.getMessage()
+                       .contains(String.format(
+                               "E11000 duplicate key error collection: %s.test index: ecm:parentId_1_ecm:name_1 dup key: { : \"%s\", : \"%s\" }",
+                               DEFAULT_MONGODB_DBNAME, domain.getId(), DOCUMENT_NAME)));
+        }
+    }
+
+    @Test
+    public void shouldFailsWhenUpdateExistingChildNameDocument() {
```

Comment:
I took a long time to understand that by "Update" you refer to the update of the internal "ecm:name" of the doc. I'd rename this to use "Move" instead of "Update".

Reply:
sorry for that i completely forgot that we cannot just update the name, and to do it we should use move and at the end of the test  i forgot to rename the test method.